### PR TITLE
feat(DataList): expose grid modifiers as properties

### LINKED ISSUE #4965: Data list - allow user to specify the breakpoint for switching between desktop and mobile layout
https://github.com/patternfly/patternfly/pull/3528

Adds `pf-m-grid` modifiers that work like the table's modifiers to switch to the mobile/grid view at a specific breakpoint.

Default behavior is the same.

* `pf-m-grid` will display the grid layout at all breakpoints
* `pf-m-grid-none` will retain the desktop layout at all breakpoints
* `pf-m-grid-[sm, md, lg, xl, 2xl]` will display the grid layout below the specified breakpoint

### PR DESCRIPTION
**Pull request by KKoukiou**
Closes #4965

### CODE REVIEW

File: packages/react-core/src/components/DataList/DataList.tsx [hunk 1]
```diff
@@ -1,8 +1,19 @@
 import * as React from 'react';
 import { css } from '@patternfly/react-styles';
 import styles from '@patternfly/react-styles/css/components/DataList/data-list';
+import stylesGrid from '@patternfly/react-styles/css/components/DataList/data-list-grid';
+import { toCamel } from '../../helpers/util';
 import { PickOptional } from '../../helpers/typeUtils';
 
+export enum DataListGridBreakpoint {
+  gridNone = 'grid-none',
```
Comment (KKoukiou) [inline packages/react-core/src/components/DataList/DataList.tsx:9]: I need to expose grid as well here 
https://github.com/patternfly/patternfly-react/issues/4965#issuecomment-704984236

**Review by dlabrecq — APPROVED** [review]

**Review by mcoker — COMMENTED** [review]
LGTM!  Thanks @KKoukiou 🥳

**Review by mcoker — DISMISSED** [review]

**Review by tlabaj — CHANGES_REQUESTED** [review]
Can you please also update the demo app and integration test to check all breakpoints.  We have had cases in the past where the breakpoints were not being applied as expected.

File: packages/react-core/src/components/DataList/DataList.tsx [hunk 2]
```diff
@@ -30,6 +42,8 @@ export interface DataListProps extends Omit<React.HTMLProps<HTMLUListElement>, '
   selectedDataListItemId?: string;
   /** Flag indicating if DataList should have compact styling */
   isCompact?: boolean;
+  /** Specifies the grid breakpoints  */
+  gridBreakPoint?: 'grid-none' | 'grid' | 'grid-sm' | 'grid-md' | 'grid-lg' | 'grid-xl' | 'grid-2xl';
```
Comment (redallen) [inline packages/react-core/src/components/DataList/DataList.tsx:46]: Change to:
```tsx
/** Modifies the data list to switch to a grid layout at a specified breakpoint */
gridBreakpoint?: 'none' | 'always' | 'sm' | 'md' | 'lg' | 'xl' | '2xl';
```

You can support `none | always` through more string transforms, or just declare this at the top of the file:
```ts
const gridBreakpointClasses = {
  'none': 'pf-m-grid', // Placeholder per https://github.com/patternfly/patternfly-react/issues/4965#issuecomment-704984236
  'always': styles.modifiers.grid,
  'sm': styles.modifiers.gridSm,
  'md': styles.modifiers.gridMd,
  'lg': styles.modifiers.gridLg,
  'xl': styles.modifiers.gridXl,
  '2xl': styles.modifiers.grid_2xl,
};
```
and then use `gridBreakPoint && gridBreakpointClasses[gridBreakPoint]`.

**Review by redallen — CHANGES_REQUESTED** [review]

File: packages/react-core/src/components/DataList/DataList.tsx [hunk 3]
```diff
@@ -30,6 +41,8 @@ export interface DataListProps extends Omit<React.HTMLProps<HTMLUListElement>, '
   selectedDataListItemId?: string;
   /** Flag indicating if DataList should have compact styling */
   isCompact?: boolean;
+  /** Specifies the grid breakpoints  */
+  gridBreakPoint?: 'none' | 'always' | 'sm' | 'md' | 'lg' | 'xl' | '2xl';
```
Comment (redallen) [inline packages/react-core/src/components/DataList/DataList.tsx:45]: Can we lowercase the `p` so it matches [OverflowMenu's spelling?](https://github.com/patternfly/patternfly-react/blob/164360d449dc64172e74159e276ed3fac2476593/packages/react-core/src/components/OverflowMenu/OverflowMenu.tsx#L24)

**Review by redallen — CHANGES_REQUESTED** [review]
Looks great!! One small nit left

**Review by redallen — APPROVED** [review]
LGTM, thanks @KKoukiou !

File: packages/react-core/src/components/DataList/DataList.tsx [hunk 4]
```diff
@@ -30,6 +41,8 @@ export interface DataListProps extends Omit<React.HTMLProps<HTMLUListElement>, '
   selectedDataListItemId?: string;
   /** Flag indicating if DataList should have compact styling */
   isCompact?: boolean;
+  /** Specifies the grid breakpoints  */
+  girdBreakpoint?: 'none' | 'always' | 'sm' | 'md' | 'lg' | 'xl' | '2xl';
```
Comment (redallen) [inline packages/react-core/src/components/DataList/DataList.tsx:45]: Not sure how I didn't catch this before... but `gird` should be `grid`.

**Review by redallen — CHANGES_REQUESTED** [review]

File: packages/react-core/src/components/DataList/DataList.tsx (hunk 4 above)
Comment (KKoukiou) [reply packages/react-core/src/components/DataList/DataList.tsx]: It's because I pushed the wrong string with the last update. thanks for catching this.

**Review by redallen — APPROVED** [review]

**Review by tlabaj — APPROVED** [review]

**Pull request merged**